include/boost/geometry/extensions/triangulation/strategies/cartesian/detail/precise_math.hpp

```diff
@@ -376,6 +314,47 @@ inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
     return(D[D_nz - 1]);
 }
 
+// see page 38, Figure 21 for the calculations, notation follows the notation in the figure.
+template
+<
+    typename RealNumber,
+    std::size_t Robustness = 3
+>
+inline RealNumber orient2d(std::array<RealNumber, 2> const& p1,
+                           std::array<RealNumber, 2> const& p2,
+                           std::array<RealNumber, 2> const& p3)
+{
+    if(Robustness == 0) {
+        return (p1[0]-p3[0])*(p2[1]-p3[1]) - (p1[1]-p3[1])*(p2[0] - p3[0]);
+    }
+    std::array<RealNumber, 2> t1, t2, t3, t4;
+    t1[0] = p1[0] - p3[0];
+    t2[0] = p2[1] - p3[1];
+    t3[0] = p1[1] - p3[1];
+    t4[0] = p2[0] - p3[0];
+    std::array<RealNumber, 2> t5_01, t6_01;
+    t5_01[0] = t1[0] * t2[0];
+    t6_01[0] = t3[0] * t4[0];
+    RealNumber det = t5_01[0] - t6_01[0];
+    RealNumber const magnitude = std::abs(t5_01[0]) + std::abs(t6_01[0]);
+
+    // see p.39, mind the different definition of epsilon for error bound
+    RealNumber const A_relative_bound =
```

Comment:
We usually don't use capitals in variable names (unless there are good reasons to do so - and maybe you have them, if they are literally on the pages you refer to)

Reply:
That is indeed the reason. All capitals in variable names in this file where chosen to mimic the notation of Shewchuk's paper as closely as possible.